ark/util/numbers.ts

```diff
@@ -34,6 +34,15 @@ export const wellFormedNumberMatcher: RegExp =
 export const isWellFormedNumber: RegExp["test"] =
 	wellFormedNumberMatcher.test.bind(wellFormedNumberMatcher)
 
+/**
+ *  Matches a well-formatted numeric expression according to the following rules:
+ *    1. Must include an integer portion (i.e. '.321' must be written as '0.321')
+ *    2. The first digit of the value must not be 0, unless the entire integer portion is 0
```

Comment:
Can we actually allow `.12` here as well? Seems pretty unambiguous and is a valid representation, just not allowed by TS.

Reply:
Done in 1d5a630